src/hooks/useSigningKeys.ts

```diff
@@ -0,0 +1,86 @@
+import { ethers } from "ethers";
+import { useState, useEffect, useContext } from "react";
+import { ErrorContext } from "common/context/ErrorContext";
+
+import { recoverPublicKey } from "features/vote/helpers/recoverPublicKey";
+import { derivePrivateKey } from "features/vote/helpers/derivePrivateKey";
+import usePrevious from "common/hooks/usePrevious";
+
+// User has to sign once per round.
+// public and private signing keys is indexed by address, and we ask them to resign if the roundMessage changes.
+export interface SigningKeys {
+  [address: string]: {
+    publicKey: string;
+    privateKey: string;
+    roundMessage: string;
+  };
+}
+
+export default function useSigningKeys(
+  signer: ethers.Signer | null,
+  address: string | null,
+  roundId: string
```

Comment:
In the calling component, it looks like we default this to `""`, which will technically trigger the if in the useEffect to be false. This works, but it's a little confusing (IMO) -- why not just allow it to be null like the others, not default it in the calling component, and assume that null means the data hasn't come in, just for the sake of consistency with the other data?

Reply:
Are you referring to roundId?